First draft Tutorials (#78)
doc-detective/doc-detective.github.io: docs/get-started/tutorials/record-video.md

---
description: Run a test that fills all fields in a form, capturing a video of the changes.
---

# Record video

This example fills in all the fields on the Watson and Holmes Intake Form, while capturing a video record.

To run the test:

> 1. Download the [TestExamples.zip](./TestExamples.zip) file and expand it on your local drive.

Reviewer: ```suggestion
1. [Set up your test environment](./set-up-your-test-environment).
```

> 2. In a terminal window, navigate to the `TestExamples` directory.
> 3. Start your local server.
> 

Reviewer: ```suggestion
2. In a terminal window, navigate to the `TestExamples` directory:

```

    ```shell

>     live-server

Reviewer: ```suggestion
    cd TestExamples
```

    ```

> 4. Enter the command 

Reviewer: ```suggestion
3. Run the _form-filler3.json_ test spec with Doc Detective:
```

   ```shell
   npx doc-detective@dev -i form-filler3.json
   ```

To record, you must run the test with _headless_ mode turned off. The display window of the browser constrains the view area of the screenshot and video functions.

## Code walkthrough

These are the notable additions to the script used in the [Capture screenshot](capture-screenshot) tutorial.

For this example, the _headless_ setting must be false, so that the test actions are visible in the browser.

```json
                    "browsers": {
                        "name": "chrome",
                        "headless": false,

>                         "window": {
>                             "width": 1200,
>                             "height": 1500
>                         },

Reviewer: ```suggestion
```

                        "viewport": {
                            "width": 1180,
                            "height": 1480
                        }
                    }
```

> Start the recording with the _record_ command and an optional output path.

Reviewer: ```suggestion
Start the recording with the [`record`](/docs/get-started/actions/record) action and an optional output path.
```

```json
                {
                    "description": "Start recording.",
                    "record": "./output/form-filler3-recording.mp4"
                },
```

After the actions are complete, stop the recording.

```json
                {
                    "description": "Stop recording.",
                    "stopRecord": true
                }
```

## Complete code sample

```json
{
    "tests": [
        {
            "runOn": [
                {
                    "platforms": [
                        "windows",
                        "mac",
                        "linux"
                    ],
                    "browsers": {
                        "name": "chrome",
                        "headless": false,

>                         "window": {
>                             "width": 1200,
>                             "height": 1500
>                         },

Reviewer: ```suggestion
```
